templates/terraform/encoders/certificate_authority.go.erb

```diff
@@ -0,0 +1,13 @@
+rc := d.Get("config.0.reusable_config.0.reusable_config").(string)
+
+parts := strings.Split(rc, "/")
+
+if len(parts) == 1 {
+    // If we have a short form: add the full path to the reusable-configs from
+    // the Google-managed project and the location of the CA.
+    config := obj["config"].(map[string]interface{})
+    configReusableConfig := config["reusableConfig"].(map[string]interface{})
+    configReusableConfig["reusableConfig"] = fmt.Sprintf("projects/568668481468/locations/%s/reusableConfigs/%s", d.Get("location"), parts[0])
```

Comment:
Just to verify: this needs to be set to a fixed project number?

Reply:
Good point. The API takes both a project number or project ID (the corresponding project ID for that project number is `privateca-data`), however the read call will always answer with the project number. So the point of passing the project number directly (instead of project ID) was to avoid a permadiff. But now that we have this custom encoder and the corresponding diff suppress function, we could pass the project ID itself since the diff suppress would only look into the suffix (the reusable config itself) when the user specifies only the short form. Would you prefer to have the project ID on the create call to make this more legible?

Reply:
Got it, I've seen that behavior in other APIs as well. I think it is fine as is, especially when the comment above clarifies the Google-managed project part. Thanks for explaining.